packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Aliases, AuthInfo } from '@salesforce/core';
+import {
+  CancelResponse,
+  ContinueResponse
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import fs = require('fs');
+import { isNullOrUndefined, promisify } from 'util';
+import { getDefaultUsernameOrAlias } from '../context/workspaceOrgType';
+import { nls } from '../messages';
+
+export interface FileInfo {
+  scratchAdminUsername?: string;
+  isDevHub?: boolean;
+  username: string;
+}
+const readFileAsync = promisify(fs.readFile);
+export class OrgList {
+  public async getAuthInfoObjects() {
+    const authFilesArray = await AuthInfo.listAllAuthFiles().catch(err => null);
+
+    if (authFilesArray === null || authFilesArray.length === 0) {
+      return null;
+    }
+    const authInfoObjects = Promise.all(
```

Comment:
We can change this from being asynchronous to synchronous if we use `fs.readFileSync`. What's the reason for having this as async ?